### PR TITLE
feat(customer): Add modals to edit a customer

### PR DESCRIPTION
**Pull request by thomas-hilaire**
## Description
- add patch customer to the API
- add modals on the customer sections to edit it

## Checklist
- [ ] The code follows the project's coding conventions and style [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

### CODE REVIEW

File: modules/web/web-app/hooks/useZodForm.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function useZodForm<TSchema extends z.ZodTypeAny>(
     name,
   })
   const withError = (name: FieldPath<z.TypeOf<TSchema>>) => ({
-    error: form.getFieldState(name)?.error?.message,
+    error: form.formState.errors[name]?.message,
```
Comment (thomas-hilaire) [inline modules/web/web-app/hooks/useZodForm.tsx:21]: an error in the form was not always triggering a rerender of the form components using `getFieldState`

File: modules/meteroid/proto/api/customers/v1/models.proto [hunk 2]
```diff
@@ -36,6 +36,16 @@ message Address {
...
 }
 
+message ShippingAddress {
+  optional string line1 = 1;
+  optional string line2 = 2;
+  optional string city = 3;
+  optional string country = 4;
+  optional string state = 5;
+  optional string zipCode = 6;
+  bool sameThanBilling = 7;
```
Comment (azhur) [inline modules/meteroid/proto/api/customers/v1/models.proto:46]: I'm thinking of another model:
```proto
message ShippingAddress {
  Address address = 1;
  bool sameAsBilling = 2;
}
``` 
WDYT?

File: modules/meteroid/proto/api/customers/v1/models.proto [hunk 3]
```diff
@@ -36,6 +36,11 @@ message Address {
   optional string zipCode = 6;
 }
 
+message ShippingAddress {
+  Address address = 1;
+  bool sameThanBilling = 7;
```
Comment (azhur) [inline modules/meteroid/proto/api/customers/v1/models.proto:41]: please rename to `sameAsBilling`

File: modules/meteroid/proto/api/customers/v1/models.proto [hunk 4]
```diff
@@ -49,5 +54,18 @@ message Customer {
   optional google.protobuf.Timestamp archived_at = 10;
   optional google.protobuf.Timestamp created_at = 11;
   optional Address billing_address = 12;
-  optional Address shipping_address = 13;
+  optional ShippingAddress shipping_address = 13;
```
Comment (azhur) [inline modules/meteroid/proto/api/customers/v1/models.proto:57]: please make sure it doesn't generate Option<Option<ShippingAddress>>, same for other fields that have custom types

Comment (thomas-hilaire) [reply modules/meteroid/proto/api/customers/v1/models.proto:57]: No it looks good, I think we can keep the optional explicit for semantic purpose

```rust
    #[prost(message, optional, tag = "12")]
    pub billing_address: ::core::option::Option<Address>,
    #[prost(message, optional, tag = "13")]
    pub shipping_address: ::core::option::Option<ShippingAddress>,
```

**Review by azhur — APPROVED** [review]

**Review by gaspb — APPROVED** [review]

**Pull request merged**